### PR TITLE
Fix as_leaflet_layer

### PR DESCRIPTION
**Pull request by davidbrochart**
Follow-up to #1033.

### CODE REVIEW

**Comment by davidbrochart** [comment]
There seems to still be an issue in the docs though.

**Comment by davidbrochart** [comment]
My bad, the docs are fine locally. I'm going to revert the changes to the docs so that it's as in #1033 @banesullivan.

File: ipyleaflet/leaflet.py [hunk 1]
```diff
@@ -1172,8 +1172,6 @@ def add(self, layer):
 
         if isinstance(layer, dict):
             layer = basemap_to_tiles(layer)
-        elif hasattr(layer, 'as_leaflet_layer'):
```
Comment (martinRenou) [inline ipyleaflet/leaflet.py:1175]: We should probably keep it here too though.

I wonder if we could implement `remove` and `substitute` too.

Comment (davidbrochart) [reply ipyleaflet/leaflet.py:1175]: Yes, let's do all that in follow-up PRs.

**Pull request merged**